app/codes/blockchain.py

```diff
@@ -44,6 +44,14 @@ def get_block(self, block_index):
 
         return block
 
+    def empty_block_hash(self, block):
+        proof = 42  # Hardcoded value for empty block
+        block_hash = self.calculate_hash(block)
+        block_hash = '0000' + block_hash[:4]
+        block['proof'] = proof
```

Comment:
this line probably needs to move ahead of block_hash = self.calculate_has(block) ?